### ISSUE TITLE
Support for custom metrics and loss functions

### ISSUE THREAD

**Issue by asdspal** [manujosephv/pytorch_tabular#42]
**Is your feature request related to a problem? Please describe.**
I would like to use a custom loss or metric function in the model. For example, here is the code for differentiable Spearman rank correlation function
```
def spearmanr(pred, target, **kw):
    pred = torchsort.soft_rank(pred, **kw)
    target = torchsort.soft_rank(target, **kw)
    pred = pred - pred.mean()
    pred = pred / pred.norm()
    target = target - target.mean()
    target = target / target.norm()
    return (pred * target).sum()
```

**Describe the solution you'd like**
The ModelConfig should accept differentiable loss functions like
```
model_config = CategoryEmbeddingModelConfig(
    task="regression",
    layers="1050-64-1",  # Number of nodes in each layer
    activation="LeakyReLU", # Activation between each layers
    learning_rate = 1e-3,
    loss = spearmanr
)
```

**Describe alternatives you've considered**

**Additional context**

**Comment by manujosephv** [comment]
@asdspal This functionality is already supported, but unfortunately you can't pass the custom loss or metrics to the `ModelConfig` because they are designed to be also initialized from yaml files.

The custom loss and metrics are only supported in the fit method, which will override any loss or metric definitions you've made in `ModelConfig`. Documentation [here](https://pytorch-tabular.readthedocs.io/en/latest/apidocs/#pytorch_tabular.tabular_model.TabularModel.fit)

Any PyTorch compatible loss function can be used here.

<html>
<body>


loss | Optional[torch.nn.modules.module.Module] | Custom Loss functions which are not in standard pytorch library | None
-- | -- | -- | --
metrics | Optional[List[Callable]] | Custom metric functions(Callable) which has the signature metric_fn(y_hat, y) and works on torch tensor inputs | None


</body>
</html>
